webapp/src/main/java/com/box/l10n/mojito/rest/textunit/CustomZonedDateTimeDeserializer.java

```diff
@@ -0,0 +1,47 @@
+package com.box.l10n.mojito.rest.textunit;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Deserialize an object into {@link ZonedDateTime}. The object can either be a number of
+ * milliseconds from 1970-01-01T00:00:00Z or a string representing a date following the ISO format,
+ * such as '2023-11-28T22:15:30'.
+ *
+ * @author hylston.barbosa
+ */
+public class CustomZonedDateTimeDeserializer extends JsonDeserializer<ZonedDateTime> {
+
+  /** logger */
+  static Logger logger = LoggerFactory.getLogger(CustomZonedDateTimeDeserializer.class);
+
+  @Override
+  public ZonedDateTime deserialize(JsonParser parser, DeserializationContext context)
+      throws IOException {
+    ZonedDateTime deserializedValue = null;
+
+    if (parser != null) {
+      try {
+        Instant instant = Instant.ofEpochMilli(parser.getLongValue());
+        deserializedValue = ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
+      } catch (JsonParseException jpe) {
+        try {
+          deserializedValue = ZonedDateTime.parse(parser.getText());
+        } catch (DateTimeParseException e) {
+          logger.warn("Unable to deserialize the date: " + parser.getText());
```

Comment:
Similar comment here in that I think we should throw a runtime exception if we cannot deserialize to a ZonedDateTime to avoid any callers continuing with their execution with a null value